### PR TITLE
Media: Use smaller images for base color generation

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary
While working https://github.com/google/web-stories-wp/pull/9905, I got better performance from blurhash with smaller images. Base color will be the same. Let's use for base color to.

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

**Comment by swissspidy** [comment]
Any before/after numbers?

**Comment by miina** [comment]
Also curious to see the before/after numbers, especially since the during the base color calculation a 10px width image is used, wondering the difference this change creates.

**Comment by spacedmonkey** [comment]
The benefits of this one seem pretty clear. My testing, I tested a 10MB image. 

#### Before 
Screenshot 2021-12-08 at 12 20 09

#### After
Screenshot 2021-12-08 at 12 22 45

The benefit here is clear. If the user has large images, like 10MB+, that is going to be slow to download and waste a lot of resources to download. 100 media items x 10MB is 1GB of data. 

The downside is also clear, 1. smallest image is less likely to already be loaded and will have to load, 2. the base color will not be as accurate, as it is a smaller sample size. 

There is benefits here, but I am willing to close this PR out, if don't think it makes sense.

**Comment by miina** [comment]
> the base color will not be as accurate, as it is a smaller sample size.

It's using a smaller image anyway when creating an image tag so that shouldn't be an issue. Also, probably accuracy is not that super important in case of this feature anyway as long as it still generally gets a similar color.

File: packages/story-editor/src/app/media/local/useContextValueProvider.js [hunk 1]
```diff
@@ -236,7 +237,8 @@ export default function useContextValueProvider(reducerState, reducerActions) {
         processVideoAudio(id, src);
       }
 
-      const imageSrc = type === 'image' ? src : poster;
+      const imageSrc =
+        type === 'image' ? getSmallestUrlForWidth(0, resource) : poster;
```
Comment (swissspidy) [inline packages/story-editor/src/app/media/local/useContextValueProvider.js:241]: Something for the future:

Do we still store `posterId` for resources?

If so, we could theoretically fetch `/wp-json/web-stories/v1/media/<posterId>` and get a smaller image size for poster images.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**